awshelper/config.go

```diff
@@ -22,79 +22,77 @@ const (
 	EnvTokenDuration   = "TERRAGRUNT_TOKEN_DURATION"
 )
 
-var sessionCache sync.Map
+var configCache sync.Map
 
-func clearSessionCache() {
-	sessionCache.Range(func(key interface{}, value interface{}) bool {
-		sessionCache.Delete(key)
+func clearConfigCache() {
+	configCache.Range(func(key interface{}, value interface{}) bool {
+		configCache.Delete(key)
 		return true
 	})
 }
 
-// CreateAwsSession returns an AWS session object for the given region, ensuring that the credentials are available
-func CreateAwsSession(awsRegion, awsProfile string) (*session.Session, error) {
-	awsSessionKey := awsRegion + "/" + awsProfile
+// CreateAwsConfig returns an AWS config object for the given region, ensuring that the credentials are available
+func CreateAwsConfig(awsRegion, awsProfile string) (*aws.Config, error) {
+	loadOptions := []func(*config.LoadOptions) error{
+		config.WithAssumeRoleCredentialOptions(func(options *stscreds.AssumeRoleOptions) {
+			options.TokenProvider = stscreds.StdinTokenProvider
+		}),
+	}
+	if awsRegion != "" {
+		loadOptions = append(loadOptions, config.WithRegion(awsRegion))
+	}
+	if awsProfile != "" {
+		loadOptions = append(loadOptions, config.WithSharedConfigProfile(awsProfile))
+	}
 
 	var (
-		awsSession *session.Session
-		err        error
+		awsConfigKey = awsRegion + "/" + awsProfile
+		awsConfig    aws.Config
+		err          error
 	)
-	if cacheValue, ok := sessionCache.Load(awsSessionKey); ok {
-		awsSession = cacheValue.(*session.Session)
+	if cacheValue, ok := configCache.Load(awsConfigKey); ok {
+		awsConfig = cacheValue.(aws.Config)
 	} else {
-		options := session.Options{
-			Profile:           awsProfile,
-			SharedConfigState: session.SharedConfigEnable,
-			AssumeRoleTokenProvider: func() (string, error) {
-				fmt.Print("Enter MFA Code: ")
-				reader := bufio.NewReader(os.Stdin)
-				mfa, err := reader.ReadString('\n')
-				return strings.TrimSpace(mfa), err
-			},
-		}
-		if awsRegion != "" {
-			options.Config = aws.Config{Region: aws.String(awsRegion)}
-		}
-		if awsSession, err = session.NewSessionWithOptions(options); err != nil {
-			return nil, errors.WithStackTraceAndPrefix(err, "Error initializing session")
+		if awsConfig, err = config.LoadDefaultConfig(context.TODO(), loadOptions...); err != nil {
```

Comment:
Out of curiosity and concerning all the `context.TODO()`, do we plan on adding cancellation propagation or something like that down the line ?